pkg/scheduler/framework/plugins/volumebinding/binder.go

```diff
@@ -247,10 +259,68 @@ func NewVolumeBinder(
 	return b
 }
 
+// GetEligibleNodes returns a list of eligible nodes by name based on the bound claims of the pod to reduce down
+// the list of nodes in subsequent scheduling stages
+func (b *volumeBinder) GetEligibleNodes(boundClaims []*v1.PersistentVolumeClaim) (sets.String, error) {
+	eligibleNodes := sets.NewString()
+	lock := sync.RWMutex{}
+	if len(boundClaims) == 0 {
+		return eligibleNodes, nil
+	}
+
+	nodes, err := b.nodeLister.List(labels.Everything())
+	if err != nil {
+		return eligibleNodes, err
+	}
+
+	var errs []error
+	workqueue.ParallelizeUntil(context.TODO(), len(nodes), len(nodes), func(i int) {
```

Comment:
It's a bit hard to prove this parallel processing across all nodes in PreFilter outweighs the processing in Filter.

I'm envisioning a way to complete this pre-check logic in O(1) (or O(logN)) time, with some help of data structure in internal cache.